runtime/fastly/host-api/host_api.cpp

```diff
@@ -127,13 +127,16 @@ size_t api::AsyncTask::select(std::vector<api::AsyncTask *> &tasks) {
   fastly::fastly_host_error err = 0;
 
   while (true) {
-    if (!convert_result(fastly::async_select(handles.data(), handles.size(),
-                                             (soonest_deadline - now) / MILLISECS_IN_NANOSECS,
-                                             &ret),
+    MOZ_ASSERT(soonest_deadline == 0 || soonest_deadline > now);
+    uint32_t timeout = soonest_deadline > 0 ? (soonest_deadline - now) / MILLISECS_IN_NANOSECS : 0;
```

Comment:
Is it worth adding a comment here the `0` means no timeout?

Reply:
Sure, how about this:

```suggestion
    // timeout value of 0 means no timeout for async_select
    uint32_t timeout = soonest_deadline > 0 ? (soonest_deadline - now) / MILLISECS_IN_NANOSECS : 0;
```

Reply:
Perfect!